### PR TITLE
Rename QueuedWorkload to Workload

### PR DESCRIPTION
**Pull request by alculquicondor**
## What type of PR is this?

/kind cleanup

/kind api-change

#### What this PR does / why we need it:

Initially we named it QueuedWorkload for lack of a better name. Now we are close to the release and:
- we didn't find a better name.
- we already referring to them as workloads in the documentation, and it works well.

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

**Comment by alculquicondor** [comment]
/hold
to prevent premature merge

**Comment by alculquicondor** [comment]
FYI @Huang-Wei. I think you wanted this name at some point :)

File: pkg/workload/workload_test.go [hunk 1]
```diff
@@ -252,15 +252,15 @@ func TestUpdateWorkloadStatus(t *testing.T) {
...
+			workload := utiltesting.MakeWorkload("foo", "bar").Obj()
 			workload.Status = tc.oldStatus
 			cl := fake.NewClientBuilder().WithScheme(scheme).WithObjects(workload).Build()
 			ctx := context.Background()
 			err := UpdateStatus(ctx, cl, workload, tc.condType, tc.condStatus, tc.reason, tc.message)
 			if err != nil {
 				t.Fatalf("Failed updating status: %v", err)
 			}
-			var updatedQW kueue.QueuedWorkload
+			var updatedQW kueue.Workload
```
Comment (ahg-g) [inline pkg/workload/workload_test.go:263]:
```suggestion
			var updatedWl kueue.Workload
```

File: docs/concepts/README.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ limits and fair sharing rules.
 A namespaced resource that groups closely related workloads belonging to a
 single tenant.
 
-### [Queued Workload](queued_workload.md)
+### [Queued Workload](workload.md)
```
Comment (ahg-g) [inline docs/concepts/README.md:18]: fix

File: pkg/workload/workload_test.go (hunk 1 above)
Comment (alculquicondor) [reply pkg/workload/workload_test.go]: Done

File: docs/concepts/README.md (hunk 2 above)
Comment (alculquicondor) [reply docs/concepts/README.md]: Done

**Comment by alculquicondor** [comment]
Rebased and squashed. Only conflict was in `scheduler.go`

**Pull request merged**